libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Schemas/Dialogs/Microsoft.QnAMakerDialog.schema

```diff
@@ -98,15 +90,22 @@
             "default": false
         },
         "rankerType": {
-            "type": "string",
-            "title": "RankerType",
+            "$ref": "https://raw.githubusercontent.com/microsoft/botframework-sdk/master/schemas/component/definitions.schema#/definitions/stringExpression",
+            "title": "Ranker Type",
             "description": "Type of Ranker.",
-            "enum": [
-                "Default",
-                "QuestionOnly",
-                "AutoSuggestQuestion"
+            "oneOf": [
+                {
+                    "enum": [
+                        "default",
+                        "questionOnly",
+                        "autoSuggestQuestion"
+                    ]
+                },
+                {
+                    "pattern": "^=(\\S){1}"
```

Comment:
>"^=(\\S){1}" [](start = 31, length = 12)

Lots of these bad patterns.

Reply:
Again, these are all by design and correc.t

---
In reply to: [410954158](https://github.com/microsoft/botbuilder-dotnet/pull/3758#discussion_r410954158) [](ancestors = 410954158)